### DISCUSSION TITLE
[Feature] Schedule on remove offline media

### DISCUSSION THREAD

**Discussion by saschabrockel** [immich-app/immich#6568] [Feature Request]
### The feature

I'm currently and I guess will continue for a long time external libraries. Changes can be frequent. It's nice to have to option to update the library with a cronjob but it does not delete media that are not existing anymore.

It would be awesome to have a cronjob option for this too.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Removing offline media automatically is risky. What if your mountpoint fails for some reason, the cronjob runs, and Immich essentially wipes your entire database unintentionally?

**Reply by saschabrockel** [reply]
Hmm not wrong though. Is there an option to trigger via an API point or something?

If I would run a cronjob outside of the container I would be sure the container runs correctly e.g. with the status healthy. I have never seen that a mountpoint fails. If the mount point would fail, Immich couldn't delete anything because the database is on this disk/mountpoint.

**Reply by jenda69** [reply]
There's this API command: [removeOfflineFiles](https://immich.app/docs/api/remove-offline-files)
You can check whether the mounts are okay before running that.
I guess you could also use something like [searchAssets](https://immich.app/docs/api/search-assets/) (/assets?isOffline=true) to check if there isn't too many offline files before running the job.
